addons/payment_adyen/models/adyen.py

```diff
@@ -81,7 +81,7 @@ def adyen_form_generate_values(self, cr, uid, id, partner_values, tx_values, con
         adyen_tx_values = dict(tx_values)
         adyen_tx_values.update({
             'merchantReference': tx_values['reference'],
-            'paymentAmount': '%d' % int(float_round(tx_values['amount'], 2) * 100),
+            'paymentAmount': '%d' % int(round(float_round(tx_values['amount'], 2) * 100)),
```

Comment:
Rounding should usually be the last thing you do after performing arithmetic operations, otherwise you're introducing new float representation errors after removing then. In other words, use `float_round(x, 0)` **after** multiplying, instead of doing `float_round(x, 2)` and then multiplying.  BTW, if you're looking for a string representation, you can use `float_repr` directly to avoid introducing more errors with additional in-between operations:
```python
'paymentAmount': float_repr(tx_values['amount'] * 100, 0),
```